### PR TITLE
DEPR: Series.argmin, argmax

### PR DESCRIPTION
**Pull request by jbrockmendel**
im looking forward to being done with these

### CODE REVIEW

File: pandas/tests/reductions/test_reductions.py [hunk 1]
```diff
@@ -695,52 +695,40 @@ def test_empty_timeseries_reductions_return_nat(self):
...
+        result = np.argmin(s)
 
-        assert result == 1
+        expected = np.argmin(data)
+        assert result == expected
 
-        with tm.assert_produces_warning(FutureWarning):
-            # argmin is aliased to idxmin
-            result = s.argmin()
+        result = s.argmin()
```
Comment (jreback) [inline pandas/tests/reductions/test_reductions.py]: how is this working? argmin/argmax are no longer defined for series

Comment (jbrockmendel) [reply pandas/tests/reductions/test_reductions.py]: its defined on IndexOpsMixin

Comment (TomAugspurger) [reply pandas/tests/reductions/test_reductions.py]: I think via IndexOpsMixin.

**Review by TomAugspurger — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**